### PR TITLE
[Arm-136] Get historical location of a friend in emergency

### PR DESCRIPTION
**Pull request by JasterV**
Handler implemented, tests to do

### CODE REVIEW

**Comment by griffobeid** [comment]
Nice refactor looks clean 🧹

File: rust/src/lang/mod.rs [hunk 1]
```diff
@@ -22,14 +22,16 @@ pub enum TranslationIds {
...
     EmergencyModePushNotificationBody,
-    PushNotificationActionView
+    PushNotificationActionView,
+    InvalidHistoricalLocationStartTime
 }
 
 
-pub fn get_dictionary(language: String) -> &'static HashMap<TranslationIds, &'static str> {
-    match language.as_str() {
+pub fn get_dictionary(language: &str) -> &'static HashMap<TranslationIds, &'static str> {
```
Comment (griffobeid) [inline rust/src/lang/mod.rs]: There is a slight difference in definition here between glossary and dictionary. glossary is like a small dictionary for a specific subject so I think that it is more appropriate for this function
```suggestion
pub fn get_glossary(language: &str) -> &'static HashMap<TranslationIds, &'static str> {
```

File: rust/src/model/telemetry.rs [hunk 2]
```diff
@@ -61,3 +63,46 @@ pub enum CommandState {
...
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("Location", 3)?;
+        s.serialize_field("data", &self.data)?;
+        s.serialize_field("device_id", &self.device_id)?;
+        s.serialize_field("timestamp", &self.timestamp.format(DATE_FORMAT).to_string())?;
+        s.end()
+    }
+}
```
Comment (griffobeid) [inline rust/src/model/telemetry.rs:85]: Can you explain what this custom Serializer is for, please :) ?

File: rust/tests/emergency_tests.rs [hunk 3]
```diff
@@ -185,3 +186,172 @@ fn test_users_state_changes_stored_correctly() {
     })
     .unwrap();
 }
+
+#[test]
+fn test_cant_get_historical_location_from_invalid_datetime_range() {
```
Comment (griffobeid) [inline rust/tests/emergency_tests.rs:191]: Nice descriptive names here 🥇

**Review by griffobeid — DISMISSED** [review]
Looks great dude 👏

File: rust/src/model/telemetry.rs (hunk 2 above)
Comment (JasterV) [reply rust/src/model/telemetry.rs:85]: @griffobeid Yeah! The NaiveDateTime type from chrono does not implement Serialize, so I can't automatically derive Serialize for the Location struct.

In addition, it allows me to customize how I want the date type value to be serialized, so I convert it to string with the format we prefer :D

File: rust/tests/emergency_tests.rs (hunk 3 above)
Comment (JasterV) [reply rust/tests/emergency_tests.rs:191]: thanks!

**Review by griffobeid — APPROVED** [review]
Ship it

**Pull request merged**